### PR TITLE
Move configure logs to PersistentPreRun

### PR DESCRIPTION
**Pull request by shamsher31**
The logging should be configured in `PreRun` steps.

### CODE REVIEW

**Comment by esnible** [comment]
Is this just code cleanup or does it change behavior?

Is this related to https://github.com/istio/istio/issues/27873 ?

**Comment by shamsher31** [comment]
> Is this just code cleanup or does it change behavior?

Just a code cleanup.

> Is this related to #27873 ?

Not related but found during debugging.

**Review by howardjohn — APPROVED** [review]

**Review by esnible — APPROVED** [review]
Code cleanup.

File: pilot/cmd/pilot-discovery/main.go [hunk 1]
```diff
@@ -56,14 +56,12 @@ var (
 	}
 
 	discoveryCmd = &cobra.Command{
-		Use:   "discovery",
-		Short: "Start Istio proxy discovery service.",
-		Args:  cobra.ExactArgs(0),
+		Use:               "discovery",
+		Short:             "Start Istio proxy discovery service.",
+		Args:              cobra.ExactArgs(0),
+		PersistentPreRunE: configureLogging,
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-discovery/main.go:62]: PreRun

File: pilot/cmd/pilot-agent/main.go [hunk 2]
```diff
@@ -175,11 +175,9 @@ var (
 			// Allow unknown flags for backward-compatibility.
 			UnknownFlags: true,
 		},
+		PersistentPreRunE: configureLogging,
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/main.go:178]: PreRun here this is not a parent command

**Pull request merged**